### ISSUE TITLE
Problems with ip-based virtual hosts?

### ISSUE THREAD

**Issue by bolera** [icing/mod_md#91]
I'm trying this interesting module for the first time and get stuck early on. I'm using the Apache 2.4.33 from Ondrej Sury, that I just installed.
It seems it doesn't like <VirtualHost 127.0.0.1:80>. It wants <VirtualHost *:80>. Is this true? I'm asking, because I'm not 100% sure, but I do not want to change all the existing vhosts just to find out if it is or not.
The problem with *:80 vhosts is that you cannot configure some domains only on certain IP addresses. All vhosts have to use all IPs the server is listening on.
Symptoms:
When using an IP I get the warning
`"No VirtualHost matches Managed Domain www.example.com". `
When I change to *:80 this message goes away but I still get
```
[Thu Jun 14 20:08:52.352905 2018] [ssl:emerg] [pid 1821:tid 140074645743488] AH01903: Failed to configure CA certificate chain!
[Thu Jun 14 20:08:52.354031 2018] [ssl:emerg] [pid 1821:tid 140074645743488] AH02311: Fatal error initialising mod_ssl, exiting. See /var/log/apache2/error.log for more information
AH00016: Configuration Failed
```
After some experimenting I **think** this happens because you cannot just add a *:80 vhost if you have other vhosts using the IP-number:80 scheme. The vhost doesn't match anything, so that the default vhost jumps in. That's surprising, I cannot see this mentioned anywhere in the apache docs. So, basically it seems you cannot mix *:80 and IP-number:80 vhosts.
And mod_md works only with *:80 vhosts.
So, before you can use it you have to change *all* vhosts to the *:80 scheme (which means you cannot have a few domains only on one IP number listening).

Is this correct or am I doing something else wrong?

If it is correct, may I suggest to make mod_md compatible with IP-number:80 based vhosts?

**Comment by bolera** [comment]
Addition. It occurred to me the problem might be the auto mode, so I set up a MDomainSet matching the ServerName and with all Aliases as an MDMember. This seems to work, e..g. (according to the ErrorLog, now set to debug) it finds a matching vhost, it creates a domain directory in domains etc. But the directory doesn't get certificates, it stays empty except for an md.json file. And no to other directory gets any content, e.g. I would expect to see an account key and some config info in accounts.
It shows a new error: "incomplete, without private key". Also, the action is stated as "update, renew", which lead me to assume it wants to renew instead of getting the cert as the md store tells it the cert was already issued. Or so.
So, I renamed the domain directory in domains. Now I get "No such file or directory", which means it wants to reuse it anyway. Where does it get this information from? I also renamed the md_store.json and I was still getting this error. So, there must be another location where this is stored and simply deleting the domain directory doesn't reinitiate the cert run. It also still says "found 1 mds in store", although I removed that store.
I then tried with the staging CA as I might have hit the rate limit. Same thing, it wants to reuse the existing domain directory. But I expect it to create a new one in the staging directory.

**Comment by bolera** [comment]
I stumbled over the utility a2md and ran "a2md --dir /etc/apache2/md --verbose drive". It seems to have a problem with contacting the LE server. However, I can just wget https://acme-staging.api.letsencrypt.org/directory fine and get a json file. This is the output:
```
[md_reg.c:191 debug][ok] md{acme1.example.com}: incomplete, without private key
[md_cmd_reg.c:277 info][ok] acme1.example.com: incomplete, sign up
[md_reg.c:893 debug][ok] acme1.example.com: run staging
[md_acme_drive.c:683 debug][ok] acme1.example.com: staging started, state=1, can_http=1, can_https=1, challenges='http-01'
[md_acme_drive.c:725 debug][ok] (null): checked creds
[md_acme.c:144 debug][ok] get directory from
[md_acme.c:419 debug][ok] req: POST
[md_curl.c:258 debug][22(Invalid argument)] request 0 failed(3): URL using bad/illegal format or missing URL
[md_acme.c:437 debug][22(Invalid argument)] req sent
[md_acme.c:158 warning][ok] unsuccessful in contacting ACME server at . If this problem persists, please check your network connectivity from your Apache server to the ACME server. Also, older servers might have trouble verifying the certificates of the ACME server. You can check if you are able to contact it manually via the curl command. Sometimes, the ACME server might be down for maintenance, so failing to contact it is not an immediate problem. mod_md will continue retrying this.
Segmentation fault
```

**Comment by bolera** [comment]
I made a2md finally work, btw, this is version 1.1.8. Documentation says that using the drive command without specifying an md should try to complete all MDs. Somehow, it's not successful with that. I'm wondering if the message "incomplete, without private key" actually refers to the account and not the md? Also, the "sign up" message that may happen right after it? This is the last error, followed by "new md ... added", that I get from md. The next error is already from ssl.
I then tried the same command but with specifying the single md that is "started". 
`a2md --dir /etc/apache2/md --acme ... --terms ... --verbose drive acme1.example.com`
